simplq/src/components/pages/QueueStatus.jsx

```diff
@@ -22,26 +23,45 @@ function QueueStatus() {
   const aheadCount = useSelector((state) => state.appReducer.aheadCount);
   const [updateInProgress, setUpdateInProgress] = useState(false);
 
-  const update = () => {
+  const showNotification = useCallback(() => {
+    const notificationImage = '/LogoLight.png';
+    const notificationText = `${queueName}: You've been notified by the queue manager.`;
+    const notification = new Notification('SimplQ', {
+      body: notificationText,
+      icon: notificationImage,
+    });
+    document.addEventListener('visibilitychange', () => {
+      if (document.visibilityState === 'visible') {
+        // The tab has become visible so clear the now-stale Notification.
+        notification.close();
+      }
+    });
+  }, [queueName]);
+
+  const update = useCallback(() => {
     clearTimeout(timeoutId);
     if (tokenId) {
+      const oldTokenStatus = tokenStatus;
       TokenService.get(tokenId)
         .then((response) => {
           dispatch(setAheadCount(response.aheadCount));
           setTokenStatus(response.tokenStatus);
+          if (response.tokenStatus === 'NOTIFIED' && oldTokenStatus === 'WAITING') {
+            showNotification();
+          }
           timeoutId = setTimeout(update, TIMEOUT);
         })
         .catch((err) => {
           handleApiErrors(err);
           timeoutId = setTimeout(update, TIMEOUT);
         });
     }
-  };
+  }, [tokenId, tokenStatus, dispatch, showNotification]);
```

Comment:
Any changes to these functions means the memoized update function has to change.
Nice way to avoid subtle bugs.